core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala

```diff
@@ -103,9 +103,10 @@ abstract class AbstractConsumerTest extends BaseRequestTest {
   }
 
   protected def sendRecords(producer: KafkaProducer[Array[Byte], Array[Byte]], numRecords: Int,
-                            tp: TopicPartition): Seq[ProducerRecord[Array[Byte], Array[Byte]]] = {
+                            tp: TopicPartition, needCurrentTime: Boolean): Seq[ProducerRecord[Array[Byte], Array[Byte]]] = {
```

Comment:
Do we need this boolean? Could we always set the current time? It seems like using the existing behavior may cause problems to many other tests too.

Reply:
Of course, I'm just afraid of some tests relying on the original "fake time". Let me test it.

Reply:
Sounds good. Yes, if some tests fail after this change, we can use the previous behavior in those tests. And the new behavior should be the default.

Reply:
OK, now, the default `sendRecord(producer, numRecords, tp)` will send with current timestamp. Some tests will verity the timestamp after consuming, so we should keep as the old behavior.